pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -100,54 +115,156 @@ func (r *ReconcileWindowsMachineConfig) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
-	// Define a new Pod object
-	pod := newPodForCR(instance)
-
-	// Set WindowsMachineConfig instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, pod, r.scheme); err != nil {
-		return reconcile.Result{}, err
-	}
-
-	// Check if this Pod already exists
-	found := &corev1.Pod{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
-	if err != nil && errors.IsNotFound(err) {
-		reqLogger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
-		err = r.client.Create(context.TODO(), pod)
+	if r.cloudProvider == nil {
+		// Get cloud provider specific info.
+		// TODO: This should be moved to validation section.
+		if instance.Spec.AWS == nil && instance.Spec.Azure == nil {
+			return reconcile.Result{}, errors.New("both the supported cloud providers are nil")
+		}
+		// As of now think of AWS implementation.
+		if instance.Spec.AWS != nil {
+			// We assume the cloud provider secret has been mounted to "~/.awscredentials` path
+			r.cloudProvider, err = cloudprovider.CloudProviderFactory(os.Getenv("KUBECONFIG"),
+				// We assume the credential path is `/etc/aws/credentials` mounted as a secret.
+				"/etc/cloud/credentials",
+				instance.Spec.AWS.CredentialAccountID,
+				"", "", instance.Spec.InstanceType,
+				// We assume the secret path is available mounted
+				instance.Spec.AWS.SSHKeyPair, "/etc/private-key.pem")
```

Comment:
Can you move these paths `/etc/private-key.pem`, `/etc/cloud/credentials` to constants?